src/tactics.ml

```diff
@@ -112,4 +112,42 @@ let handle_tactic : sig_state -> Proof.t -> p_tactic -> Proof.t =
   | P_tac_refl          ->
       handle_refine (Rewrite.reflexivity tac.pos ps)
   | P_tac_sym           ->
-      handle_refine (Rewrite.symmetry tac.pos ps)
+    handle_refine (Rewrite.symmetry tac.pos ps)
+  | P_tac_why3(s)    ->
+      (* get the goal to prove *)
+      let (hypothesis, trm) = Proof.Goal.get_type g in
+      let trm = unfold trm in
+      (* get the default or the indicated name of the prover. *)
+      let prover_name =
+        match s with
+        | None          -> Timed.(!Why3prover.current_prover)
+        | Some(name)    -> name.elt
+      in
+      (* translate from lambdapi to why3 terms. *)
+      let (l_prop, hyps, why3term) =
+          Why3prop.translate ps.proof_builtins (hypothesis, trm) in
+      (* create a new task that contains symbols, axioms and the goal. *)
+      let tsk = Why3task.create l_prop hyps why3term in
+      (* call the prover named [prover_name] and get the result. *)
+      let prover_result = Why3prover.call prover_name tsk in
+      (* if the prover succeed to prover the goal. *)
+      if Why3prover.answer prover_result.pr_answer then
+        (* create a new axiom that represents the proved goal. *)
+        let why3_axiom = Pos.none (Why3prop.get_newname ()) in
```

Comment:
Perhaps, we could use tac.pos instead of Pos.none?

Reply:
I've used `Pos.make` instead in 58922873d7f03b4e68f7c283a56c001fc94f8c4a